plugin/ocgrpc/trace_common.go

```diff
@@ -55,26 +47,32 @@ func (c *clientTraceHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo)
 // it finds one, uses that SpanContext as the parent context of the new span.
 //
 // It returns ctx, with the new trace span added.
-func (s *serverTraceHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo) context.Context {
+func (s *ServerHandler) traceTagRPC(ctx context.Context, rti *stats.RPCTagInfo) context.Context {
 	md, _ := metadata.FromIncomingContext(ctx)
 	name := "Recv" + strings.Replace(rti.FullMethodName, "/", ".", -1)
-	if s := md[traceContextKey]; len(s) > 0 {
-		if parent, ok := propagation.FromBinary([]byte(s[0])); ok {
-			ctx, _ = trace.StartSpanWithRemoteParent(ctx, name, parent, trace.StartOptions{})
-			return ctx
+	options := trace.StartOptions{Sampler: s.Sampler}
+	traceContext := md[traceContextKey]
+	if !s.NoReadTraceContext && len(traceContext) > 0 {
+		// Metadata with keys ending in -bin are actually binary. They are base64
+		// encoded before being put on the wire, see:
+		// https://github.com/grpc/grpc-go/blob/08d6261/Documentation/grpc-metadata.md#storing-binary-data-in-metadata
```

Comment:
Where are we checking that the keys end in `-bin`? If we aren't, perhaps let's update this comment.

Reply:
`traceContextKey` ends in `-bin` - it is a constant

Reply:
It's just a little confusing and without context for me as a reader that looked at this PR, without the mention of what uses `-bin`.

Reply:
Oh I see where you're coming from. The reason I added it was that I was confused by why we were just assuming that the value of the metadata (which is typed as string) is actually the raw bytes and not hex encoded (as, I would argue, the type would suggest). So I added this documentation to clarify the situation. I didn't introduce the use of -bin in this PR.